src/compat-cloexec.c

```diff
@@ -0,0 +1,101 @@
+#include "compat-cloexec.h"
+
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef _POSIX_C_SOURCE
```

Comment:
@ensc Could you please check:
```c
#ifdef _POSIX_C_SOURCE
#  include <fcntl.h>
```
In msys2 mingw-w64 builds we see these warnings e.g. [1]:
```
compat-cloexec.c:11: warning: "O_CREAT" redefined
   11 | #  define O_CREAT       0
      | 
In file included from ./rrd_config_bottom.h:64,
                 from ./rrd_config.h:634,
                 from compat-cloexec.h:4,
                 from compat-cloexec.c:1:
D:/a/_temp/msys64/mingw64/include/fcntl.h:40: note: this is the location of the previous definition
   40 | #define O_CREAT _O_CREAT
      | 
compat-cloexec.c:12: warning: "O_WRONLY" redefined
   12 | #  define O_WRONLY      0
      | 
D:/a/_temp/msys64/mingw64/include/fcntl.h:37: note: this is the location of the previous definition
   37 | #define O_WRONLY _O_WRONLY
      | 
compat-cloexec.c:13: warning: "O_RDONLY" redefined
   13 | #  define O_RDONLY      0
      | 
D:/a/_temp/msys64/mingw64/include/fcntl.h:36: note: this is the location of the previous definition
   36 | #define O_RDONLY _O_RDONLY
      | 
compat-cloexec.c:14: warning: "O_APPEND" redefined
   14 | #  define O_APPEND      0
      | 
D:/a/_temp/msys64/mingw64/include/fcntl.h:39: note: this is the location of the previous definition
   39 | #define O_APPEND _O_APPEND
      | 
compat-cloexec.c:15: warning: "O_RDWR" redefined
   15 | #  define O_RDWR        0
      | 
D:/a/_temp/msys64/mingw64/include/fcntl.h:38: note: this is the location of the previous definition
   38 | #define O_RDWR _O_RDWR
      | 
```
**`configure`** says:
```
checking for fcntl.h... yes
```
So, `HAVE_FCNTL_H` is defined.

[1] https://github.com/oetiker/rrdtool-1.x/actions/runs/4740218558/jobs/8415805010#step:5:901